src/pages/AppDetail.svelte

```diff
@@ -9,12 +12,21 @@
   import Pill from "../components/Pill.svelte";
   import { TabGroup, Tab, TabContent } from "../components/tabs";
   import PageTitle from "../components/PageTitle.svelte";
-  import { pageTitle } from "../state/stores";
+  import { pageState, pageTitle } from "../state/stores";
 
   export let params;
 
   const appDataPromise = getAppData(params.app);
 
+  let itemType = $pageState.itemType || "metrics";
+  const searchText = writable($pageState.search || "");
+  setContext("searchText", searchText);
+  const showExpired = writable($pageState.showExpired || false);
+  setContext("showExpired", showExpired);
```

Comment:
If `pageState` store already tracks search, itemType, showExpired, (and maybe more that I'm missing), why is it necessary to create context for these properties? Aren't `searchText` and `$pageState.search` always referring to the same value? 

> There is top-level persistence handling in App.svelte which uses a "pageState" store, that then propagates down to components. This allows us to define persistence inside each page in a relatively uniform way.

If ^ is the underlying principle to handle states in these pages, wouldn't it be simpler to retrieve and update the states directly from the store, instead of creating another access layer of context on top? 

What I'm trying to say is, wherever we need to use state variables, we could just call directly from the store. And same goes for when we need to update them. I feel like using context for a store property is a bit redundant, but I could be missing something major here.

Reply:
I think you're right. I think my reasoning was that we should use context to make the components more reusable, but in practice it just makes things more complicated and these components are pretty context-dependent in any case.

I'll give what you suggest a try.

Reply:
Ok, I dug more into this. 

I think, unfortunately, that something like the current approach is probably about the best we can do if we want to continue making the tabs a reusable component (rather than something that is tightly coupled to a notion of displaying an item list in the application page, which is what we're currently using it for).

The most promising place to clean stuff up would be TabGroup.svelte, which uses its own store as well as context. It would seem logical to just pass in a store and have it manipulate that (and it's totally possible), but that makes testing it as a standalone component and forces you to use it as a store, which I think is not necessarily what we'd want in all cases. I gave this a try in but reverted it after realizing that it broke the stories, you can see what it looks like here: https://github.com/mozilla/glean-dictionary/pull/512/commits/a90db1457fc9575a8d678cf12fbd4fe78205f092